### ISSUE TITLE
feature request: current day

### ISSUE THREAD

**Issue by NathanS-8500** [Peppie84/FS22_ExtendedGameInfoDisplay#3]
Hey, Peppie84! I like what you did with this mod! No issues and love it. As the title suggests, would it be possible to add a feature of this mod that allows it to get and read the current day information from the environment xml of any gamesave that is created and display say

<currentDay>1</currentDay> shows

Current Day: 1

just above the temp readings.

<currentDay>25</currentDay> would show

Current Day: 25 on HUD.

Basically a display that displays the current day information on the HUD. I would love this mod even better if this were added, because it would save me the hassle of trying to remember what career day it is currently without having to save my game, minimize it, and look at that information manually in the environment.xml file.

**Comment by Peppie84** [comment]
Hi @NathanS-8500 ,

thanks for the flowers. To your feature request: Can you tell me the benefit you have by reading the `currentDay` value from the evironment.xml of your save?

I just started a new save and the value is `currentDay=6`.

What i can tell:
FS22 use this value to calculate the current day + current month. By a new save game, it starts with `6` means (1 day per month settings)

- Mar=1,
- Apr=2,
- May=3
- Jun=4,
- Jul=5,
- **_Aug=6_**
- Sep=7
- Oct=8
- Nov=9
- Dec=10
- Jan=11
- Feb=12

 Every new save starts with August. If you switch the daysPerPeriod=2 and fast forward to the next month, the new `currentDay` switches to 14.

That means

> `roundOff(14 / 2) = 7 = Sep ((14 modulo 2) + 1) = Sep 1`

one day ahead, the current day is 15, stands for:

> `roundOff(15 / 2) = 7 = Sep ((15 modulo 2) + 1) = Sep 2`

another day ahead, the current day is 16, stands for:

> `roundOff(16 / 2) = 8 = Oct ((16 modulo 2) + 1) = Oct 1`

Edit: Even the currentYear is just calculated by the `currentDay` value. There is no `currentYear` on the environment.xml.

**Comment by NathanS-8500** [comment]
Peppie84, from what I've seen of the currentDay value in my games, it has nothing to do with seasons whether the seasons are turned on or off. Throughout my gameplay and checking this value out, this currentDay value will continue to climb beyond 16 if the player plays more than 16 currentDays. I’ve seen that value toward 50 or more already and it only means how many days in-game have passed. Every time midnight hits, that currentDay value goes up a digit to reflect the next career day. My only benefit from this would be to see how many days have passed in game without having to look at the environment.xml file manually. Having this HUD implemented would allow for the mod to get, read, and display that value on the HUD in game for us players that are interested in seeing how many days have passed in-game. It would also allow for players with seasons to check that and see when the next month will display on the HUD depending on the season settings. In short, a HUD to display the currentDay value it reads and gets from environment.xml.

**Comment by Peppie84** [comment]
Hi @NathanS-8500 ,
thanks for your request but i think it has currently no value for this mod.

**_Edit:_** As a workaround, you can calculate the value by this formula: `((currentYear-1)*12 + MonthInNumberFromTheListAbove)` if you play with 1 DayPerMonth otherwise use the formula above i showed you.
![image](https://user-images.githubusercontent.com/7722222/234404371-e4e0f2b8-3e1c-4939-8891-968a38957c63.png)
For this example: `(5-1)*12 + 4 (for Jun)`=`52`
